### ISSUE TITLE
Not able to edit PDF files.

### ISSUE THREAD

**Issue by ritulahkar** [JakubMelka/PDF4QT#196]
When trying to edit the text of the following file with the latest appimage i.e.: PDF4QT-1.4.0.0-x86_64.AppImage, it fails and shows the following error. 

![image](https://github.com/user-attachments/assets/3d489d3e-147d-4467-b6bb-b3ce39377c3f)

The following is the file:
[sample.pdf](https://github.com/user-attachments/files/16182699/sample.pdf)

I am using Linux mint 21.3.

**Comment by remenor** [comment]
I would add that no text edition is saved in the file. After saving the changes or creating a new file with the changes, the text remains the same (using Appimage)

**Comment by AlisterH** [comment]
> I would add that no text edition is saved in the file. After saving the changes or creating a new file with the changes, the text remains the same (using Appimage)

Yes, I see the same thing, in my case I tested with the Windows zip package.

But this is a different issue, as it seems to occur with all files, not just ones that produce an error like in the screenshot.

**Comment by AlisterH** [comment]
With regard to the issue in the screenshot, I can create a test pdf a certain way, such that I get this error when editing the text to insert a character that wasn't in the original pdf, but not if I only reuse characters that were in the original pdf.  I guess this is because the pdf has only embedded a subset of the font, and pdf4qt doesn't know how to find and embed a new character, so this may be expected behaviour.
But in the case of ritulahkar's sample.pdf I get the error even if I don't introduce a new character.

**Comment by AlisterH** [comment]
OK, at least with my testing, the problem is not edits failing to save, but edits being saved incorrectly.

> > I would add that no text edition is saved in the file. After saving the changes or creating a new file with the changes, the text remains the same (using Appimage)
>
> Yes, I see the same thing, in my case I tested with the Windows zip package.

I find that if I add a "text label", "accept mark", "reject mark", or "svg image", then they are saved, but incorrectly - they are at or outside the bottom left corner of the page.  This may have something to do with the geometry/transformation of the page.
Here I draw some stuff on a blank page:
![image](https://github.com/user-attachments/assets/2c1cd429-4d0e-439c-879b-903973042e2b)
When I toggle off "edit page content", text labels etc seem to disappear.  But they are actually down at the bottom left corner.
![image](https://github.com/user-attachments/assets/94305136-d34e-4c0c-8340-b19d929da313)
And if I toggle "edit page content" back on then I can select and move them (as long as they aren't completely off the edge of the page).
Also, the text label is mangled (wrong size and letters flipped and out of order):
![image](https://github.com/user-attachments/assets/9a81afcc-1fd2-4758-9548-b4118bd2feb7)

Here is the test file I'm drawing on: 
[combined-20240903-1405-PID39.pdf](https://github.com/user-attachments/files/16842745/combined-20240903-1405-PID39.pdf)

I wonder if this could be related to #203

**Comment by AlisterH** [comment]
The other thing that is likely to confuse people is that you can save the file without exiting the "edit page content mode", in which case the saved file won't have your edits, and this is not obvious.

There are a few options to solve this problem:
- "Save" and "save as" could be disabled while you are in "edit page content mode", or
- "When you save while in the "edit page content mode" there could be a warning message saying that edits which haven't been "written to the document" will not be saved to file, or
- It could offer to write those changes to the document and save.

Perhaps some changes to terminology would also be helpful - e.g. changes which haven't been "written to the document" could be called "uncommitted" or "pending".

**Comment by JakubMelka** [comment]
@ritulahkar, @AlisterH, @remenor, I have fixed all these issues.
